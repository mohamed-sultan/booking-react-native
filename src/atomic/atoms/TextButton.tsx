import React from 'react';
import {TouchableOpacity, Text, StyleSheet} from 'react-native';
import  { useThemeColors } from '@store/useTheme';

const TextButton = ({label, selected = false, onPress}: any) => {
  const colors = useThemeColors();

  const styles = getStyles(colors);
  return (
    <TouchableOpacity
      style={[styles.button, selected && styles.selectedButton]}
      onPress={onPress}>
      <Text style={styles.text}>{label}</Text>
    </TouchableOpacity>
  );
};
export default React.memo(TextButton);

const getStyles = (colors: any) => StyleSheet.create({
  button: {
    paddingVertical: 16,
    paddingHorizontal: 24,
    borderWidth: 1,
    borderColor: colors.LightBackground,
    borderRadius: 8,
    marginVertical: 8,
  },
  selectedButton: {
    borderColor: colors.PrimaryBackground,
  },
  text: {
    fontSize: 16,
    textAlign: 'left',
    color: colors.PrimaryText,
  },
});


