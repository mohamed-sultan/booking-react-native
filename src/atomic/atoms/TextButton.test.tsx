import React from 'react';
import renderer from 'react-test-renderer';
import TextButton from './TextButton';

// Mock the theme hook
jest.mock('@store/useTheme', () => ({
  useThemeColors: () => ({
    LightBackground: '#ccc',
    PrimaryBackground: '#007bff',
    PrimaryText: '#000',
  }),
}));

describe('TextButton', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <TextButton label="Click Me" onPress={() => {}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when selected', () => {
    const tree = renderer.create(
      <TextButton label="Selected" selected={true} onPress={() => {}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
