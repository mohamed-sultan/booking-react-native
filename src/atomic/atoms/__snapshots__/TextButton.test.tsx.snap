// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton renders correctly when selected 1`] = `null`;

exports[`TextButton renders correctly with default props 1`] = `null`;
